src/prefect/client/orchestration.py

```diff
@@ -3329,6 +3330,32 @@ async def read_resource_related_automations(
     async def delete_resource_owned_automations(self, resource_id: str):
         await self._client.delete(f"/automations/owned-by/{resource_id}")
 
+    async def api_version(self) -> str:
+        res = await self._client.get("/admin/version")
+        return res.json()
+
+    def client_version(self) -> str:
+        return prefect.__version__
+
+    async def api_compatible(self):
+        # Cloud is always compatible as a server
+        if self.server_type == ServerType.CLOUD:
+            return
+
+        try:
+            api_version = await self.api_version()
+        except Exception as e:
+            raise RuntimeError(f"Failed to reach API at {self.api_url}") from e
+
+        api_version = version.parse(api_version)
+        client_version = version.parse(self.client_version())
+
+        if api_version.major != client_version.major:
+            raise RuntimeError(
+                f"Client version {client_version} is incompatible with API version {api_version}. "
```

Comment:
Could use some feedback here and below if we want to use `API`, `Server` or other just to keep things consistent

Reply:
i think the "API version" is kind of confusing and should be more of an implementation detail - I don't think it helps a user know what to do next. So that means reporting the prefect version on the server is probably more actionable / useful IMO

Reply:
Makes sense, updated the error message accordingly.